src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -90,7 +94,7 @@ public function supportsNormalization($data, $format = null)
      */
     public function denormalize($data, $type, $format = null, array $context = [])
     {
-        $dateTimeFormat = $context[self::FORMAT_KEY] ?? null;
+        $dateTimeFormat = $context[self::FORMAT_KEY] ?? $this->defaultDenormalizationFormat;
```

Comment:
why not `$this->defaultContext[self::FORMAT_KEY] ?? null;` here instead of the property?

Reply:
That was my first idea, but the default value of `$this->defaultContext[self::FORMAT_KEY]` == `\DateTime::RFC3339` when not passed as an argument to the constructor, and that makes `\DateTime::createFromFormat($dateTimeFormat, $data, $timezone)` fail with common formats such as 'Y-m-d' (this would cause a BC Break, as well as cause multiple other tests to fail).

When the format is not passed as an argument to the constructor, we want the method to return `new \DateTime($data, $timezone)`, and therefore we need `$dateTimeFormat` to equal null. 

That default value set in the constructor forces me to store the argument passed one, and that's why i added the `$defaultDenormalizationFormat` property.

I could add a comment at the variable assignation explaining that if you want

Reply:
So, what should I do ?